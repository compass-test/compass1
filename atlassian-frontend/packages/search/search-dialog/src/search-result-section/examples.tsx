import React from 'react';
import { SearchResultSection } from './search-result-section';
import times from 'lodash/times';

const dummyResultItemStyling = {
  padding: '0 8px',
};

export const Basic = () => (
  <SearchResultSection title="Search Results" totalResults={12}>
    {
      // TODO: use SearchResult component here when it is implemented
      times(10, () => (
        <div style={dummyResultItemStyling}>search result</div>
      ))
    }
  </SearchResultSection>
);

export const WithoutTotalResultsBadge = () => (
  <SearchResultSection title="Search Results">
    {
      // TODO: use SearchResult component here when it is implemented
      times(10, () => (
        <div style={dummyResultItemStyling}>search result</div>
      ))
    }
  </SearchResultSection>
);

export default { title: 'Search Dialog/Search Results Section' };
