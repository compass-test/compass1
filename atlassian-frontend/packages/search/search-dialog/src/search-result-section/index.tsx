export { SearchResultSection } from './search-result-section';
