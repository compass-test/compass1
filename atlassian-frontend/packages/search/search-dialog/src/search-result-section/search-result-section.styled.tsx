import styled from '@emotion/styled';
import { N200 } from '@atlaskit/theme/colors';
import { grid } from '../style-utils';

export const SearchResultSectionContainer = styled.div`
  padding: ${grid.multiple(1).px} 0;
`;

export const SearchResultSectionHeading = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-bottom: 6px;
  max-height: ${grid.multiple(4).px};
`;

export const SearchResultsSectionTitle = styled.span`
  font-size: 11px;
  line-height: ${grid.twice().px};
  font-weight: 600;
  text-transform: uppercase;
  margin: 0 ${grid.twice().px};
  color: ${N200};
`;
