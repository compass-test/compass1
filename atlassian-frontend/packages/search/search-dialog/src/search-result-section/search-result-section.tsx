import React from 'react';
import Badge from '@atlaskit/badge';
import {
  SearchResultsSectionTitle,
  SearchResultSectionContainer,
  SearchResultSectionHeading,
} from './search-result-section.styled';

export interface Props {
  /**
   * The title of the section
   */
  title: string | JSX.Element;
  /**
   * If provided this will show the total number of results next to the title.
   */
  totalResults?: number;
}

export const SearchResultSection: React.FunctionComponent<Props> = ({
  title,
  totalResults,
  children,
}) => {
  return (
    <SearchResultSectionContainer>
      <SearchResultSectionHeading>
        <SearchResultsSectionTitle>{title}</SearchResultsSectionTitle>
        {totalResults ? (
          <Badge testId="search-result-count-badge">{totalResults}</Badge>
        ) : null}
      </SearchResultSectionHeading>
      {children}
    </SearchResultSectionContainer>
  );
};
