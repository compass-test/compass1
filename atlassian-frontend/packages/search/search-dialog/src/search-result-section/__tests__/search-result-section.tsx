import React from 'react';
import { SearchResultSection } from '../search-result-section';
import { shallow } from 'enzyme';
import Badge from '@atlaskit/badge';

jest.mock('@atlaskit/badge');

describe('<SearchResultSection />', () => {
  it('snapshot', () => {
    const wrapper = shallow(
      <SearchResultSection title="Search results" totalResults={22} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('snapshot when totalResults = 0', () => {
    const wrapper = shallow(
      <SearchResultSection title="Search results" totalResults={0} />,
    );
    expect(wrapper.find(Badge).exists()).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it('snapshot when totalResults is undefined', () => {
    const wrapper = shallow(
      <SearchResultSection title="Search results" totalResults={0} />,
    );
    expect(wrapper.find(Badge).exists()).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });
});
