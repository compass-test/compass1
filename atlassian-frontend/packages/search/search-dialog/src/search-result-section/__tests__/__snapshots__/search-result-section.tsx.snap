// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultSection /> snapshot 1`] = `
<Styled(div)>
  <Styled(div)>
    <Styled(span)>
      Search results
    </Styled(span)>
    <Badge
      testId="search-result-count-badge"
    >
      22
    </Badge>
  </Styled(div)>
</Styled(div)>
`;

exports[`<SearchResultSection /> snapshot when totalResults = 0 1`] = `
<Styled(div)>
  <Styled(div)>
    <Styled(span)>
      Search results
    </Styled(span)>
  </Styled(div)>
</Styled(div)>
`;

exports[`<SearchResultSection /> snapshot when totalResults is undefined 1`] = `
<Styled(div)>
  <Styled(div)>
    <Styled(span)>
      Search results
    </Styled(span)>
  </Styled(div)>
</Styled(div)>
`;
